src/Interpreters/PartLog.cpp

```diff
@@ -233,4 +253,28 @@ bool PartLog::addNewParts(
     return true;
 }
 
+bool PartLog::addNewPart(ContextPtr context, const PartLog::PartLogEntry & part, const ExecutionStatus & execution_status)
+{
+    return addNewParts(context, {part}, execution_status);
+}
+
+bool PartLog::addNewParts(ContextPtr context, const PartLog::MutableDataPartsVector & parts, UInt64 elapsed_ns,
+                 const ExecutionStatus & execution_status)
+{
+    PartLog::PartLogEntries part_log_entries;
+    part_log_entries.reserve(parts.size());
+
+    for (const auto & part : parts)
+        part_log_entries.emplace_back(part, elapsed_ns);
+
+    return addNewParts(context, part_log_entries, execution_status);
+}
+
+bool PartLog::addNewPart(ContextPtr context, const PartLog::MutableDataPartPtr & part, UInt64 elapsed_ns, const ExecutionStatus & execution_status)
+{
+    PartLog::PartLogEntries part_log_entries;
+    part_log_entries.emplace_back(part, elapsed_ns);
+    return addNewParts(context, part_log_entries, execution_status);
+}
```

Comment:
These overloads should be removed. I'll check where else they're used and replace them with new ones.